src/createNativeStackNavigator.tsx

```diff
@@ -330,6 +331,41 @@ class StackView extends React.Component<Props> {
     return <ScreenStackHeaderConfig {...headerOptions} />;
   };
 
+  private maybeRenderNestedStack = (
+    isHeaderInModal: boolean,
+    screenProps: unknown,
+    route: NavigationRoute<NavigationParams>,
+    navigation: NavigationScreenProp<
+      NavigationRoute<NavigationParams>,
+      NavigationParams
+    >,
+    SceneComponent: React.ComponentType<Record<string, unknown>>,
+    index: number,
+    descriptor: NativeStackDescriptor
+  ) => {
+    if (isHeaderInModal) {
```

Comment:
This may be problematic if this option changes dynamically as it'll unmount the previous component and render new one, and all state will be destroyed.